www/activities/diary/js/diary.js

```diff
@@ -170,6 +170,33 @@ app.Diary = {
     el.innerText = dateString;
   },
 
+  sendStatistics: async function () {
+    let address = app.Settings.get("developer", "data-sharing-address");
+    let wifiOnly = app.Settings.get("developer", "data-sharing-wifi-only");
+
+    if (app.Settings.get("developer", "data-sharing-active") == true && !!address
```

Comment:
This condition expression is incorrect! Since `&&` binds stronger than `||`, the statement evaluates to `true` even when data sharing is disabled and no address is set. It suffices that `wifiOnly` is `false` for the whole statement to become `true`.

To fix this and to make the code more readable, I'd suggest to change it to:

```js
if (app.Settings.get("developer", "data-sharing-active") == true && !!address) {
  if ((wifiOnly && navigator.connection.type == "wifi") || !wifiOnly) {
    ...
  }
}
```

Reply:
will change it. But I guess this worked because of the double check. but not sure.
should not work tbh.

because of this I normaly work like
```
if (app.Settings.get("developer", "data-sharing-active") == false) {
   return;
}
if(!address) {
   return;
}
if (navigator.connection.type == "wifi" || !wifiOnly) {
    return;
}
```
In my opinion its better for understanding.
But I will change it to your suggestions.

Reply:
done